backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java

```diff
@@ -576,12 +576,21 @@ private ApiUser getCurrentApiUserNoCache() {
     return nonOktaUser.orElseGet(() -> getCurrentApiUserFromIdentity(userIdentity));
   }
 
+  /*
+   `getCurrentUserInfoForWhoAmI()` can be removed and replaced with `getCurrentUserInfo()` as part of #7602 or whenever we stop migrating users over from Okta
+  */
   public UserInfo getCurrentUserInfoForWhoAmI() {
     ApiUser currentUser = getCurrentApiUser();
+
     Optional<OrganizationRoles> currentOrgRoles = _orgService.getCurrentOrganizationRoles();
     boolean isAdmin = _authService.isSiteAdmin();
-    if (!_featureFlagsConfig.isOktaMigrationEnabled() && !isAdmin) {
-      setRolesAndFacilities(currentOrgRoles, currentUser);
+    if (!isAdmin) {
+      if (currentOrgRoles.isPresent()) {
+        PartialOktaUser oktaUser = _oktaRepo.findUser(currentUser.getLoginEmail());
+        return consolidateUser(currentUser, oktaUser);
```

Comment:
⚠️ Making use of the [`consolidateUser` method](https://github.com/CDCgov/prime-simplereport/blob/119c6cca47d2f450e307cf551b1e800f82663db3/backend/src/main/java/gov/cdc/usds/simplereport/service/ApiUserService.java#L716) that also handles the comparison check and logging